transformer_engine/common/util/cast_kernels.cuh

```diff
@@ -248,11 +248,12 @@ __global__ void __launch_bounds__(MXFP8_THREADS_PER_CHUNK)
             const bool out_of_bounds = (col_out_of_bounds || row_out_of_bounds);
 
             float elt = static_cast<float>(in.data.elt[j]);
-            if constexpr (IS_ACT || IS_DACT) {
+            if constexpr (IS_ACT) {
               elt = OP(elt, {});
             }
```

Comment:
So if I understand correctly, this is the bug we're trying to fix. If the forward pass is `y = f(x)`, we were previously computing `dx = x * df(dy)` instead of `dx = dy * df(x)`.

Reply:
That's correct.